code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/builders/FilmstripCarouselNotificationBuilder.kt

```diff
@@ -0,0 +1,288 @@
+import androidx.core.app.NotificationCompat
+import com.adobe.marketing.mobile.core.R
+import com.adobe.marketing.mobile.services.Log
+import com.adobe.marketing.mobile.services.ServiceProvider
+import com.adobe.marketing.mobile.services.caching.CacheService
+import com.adobe.marketing.mobile.services.ui.notification.CarouselTemplateUtil
+import com.adobe.marketing.mobile.services.ui.notification.NotificationConstructionFailedException
+import com.adobe.marketing.mobile.services.ui.notification.PushTemplateConstants
+import com.adobe.marketing.mobile.services.ui.notification.PushTemplateImageUtil
+import com.adobe.marketing.mobile.services.ui.notification.models.CarouselPushTemplate
+import com.adobe.marketing.mobile.services.ui.notification.models.ManualCarouselPushTemplate
+
+/**
+ * Object responsible for constructing a [NotificationCompat.Builder] object containing a manual filmstrip carousel template notification.
+ */
+internal object FilmstripCarouselNotificationBuilder {
+    private const val SELF_TAG = "FilmstripCarouselTemplateNotificationBuilder"
+
+    @Throws(NotificationConstructionFailedException::class)
+    fun construct(
+        context: Context,
+        pushTemplate: ManualCarouselPushTemplate?,
+        trackerActivityClass: Class<out Activity>?,
+        broadcastReceiverClass: Class<out BroadcastReceiver>?,
+    ): NotificationCompat.Builder {
+        if (pushTemplate == null) {
+            throw NotificationConstructionFailedException(
+                "push template is null, cannot build a manual filmstrip carousel template notification."
+            )
+        }
+        val cacheService = ServiceProvider.getInstance().cacheService
+            ?: throw NotificationConstructionFailedException(
+                (
+                    "Cache service is null, manual filmstrip carousel push notification will not be" +
+                        " constructed."
+                    )
+            )
+        Log.trace(
+            PushTemplateConstants.LOG_TAG,
+            SELF_TAG,
+            "Building a manual filmstrip carousel template push notification."
+        )
+
+        val packageName = context.packageName
+        val smallLayout = RemoteViews(packageName, R.layout.push_template_collapsed)
+        val expandedLayout = RemoteViews(packageName, R.layout.push_template_filmstrip_carousel)
+
+        // create a silent notification channel if needed
+        if (pushTemplate.isFromIntent == true && Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            val notificationManager =
+                context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+            AepPushNotificationBuilder.setupSilentNotificationChannel(
+                notificationManager,
+                pushTemplate.getNotificationImportance()
+            )
+        }
+
+        // create the notification channel if needed
+        val channelIdToUse = AepPushNotificationBuilder.createChannel(
```

Comment:
Should this be inside `else` block so as to not replace the silent channel created on line 73?

Reply:
the silent channel creation has been moved out of the call to `createChannel`